src/poetry/repositories/pool.py

```diff
@@ -1,59 +1,78 @@
 from __future__ import annotations
 
+from collections import OrderedDict
+from dataclasses import dataclass
+from enum import IntEnum
+from enum import auto
 from typing import TYPE_CHECKING
 
+from poetry.repositories.abstract_repository import AbstractRepository
 from poetry.repositories.exceptions import PackageNotFound
-from poetry.repositories.repository import Repository
 
 
 if TYPE_CHECKING:
     from poetry.core.constraints.version import Version
     from poetry.core.packages.dependency import Dependency
     from poetry.core.packages.package import Package
 
+    from poetry.repositories.repository import Repository
 
-class Pool(Repository):
+
+class Priority(IntEnum):
+    # The order of the members below dictates the actual priority. The first member has
+    # top priority.
+    DEFAULT = auto()
+    PRIMARY = auto()
+    SECONDARY = auto()
```

Comment:
```suggestion
    DEFAULT = enum.auto()
    PRIMARY = enum.auto()
    SECONDARY = enum.auto()
```

Very minor nit, but I prefer `enum.auto()` here as `auto()` is a very generic function name.